flink/v1.15/flink/src/test/java/org/apache/iceberg/flink/source/TestFlinkLookupFunction.java

```diff
@@ -0,0 +1,258 @@
+
+    env.execute("Test Iceberg DataStream");
+
+    flinkLookupFunction.open(null);
+    flinkLookupFunction.setCollector(
+        new Collector<RowData>() {
+          @Override
+          public void collect(RowData record) {
+            Assert.assertEquals(
+                "Should join the table data by key correctly. "
+                    + "The joined 'id' value should be '1'",
+                expectedRowData.getInt(0),
+                record.getInt(0));
+            Assert.assertEquals(
+                "Should join the table data by key correctly. "
+                    + "The joined 'data' value should be 'hello'\"",
+                expectedRowData.getString(1),
+                record.getString(1));
+          }
+
+          @Override
+          public void close() {}
+        });
+    flinkLookupFunction.eval(key);
+  }
+
+  @Test
+  public void testFlinkLookupFunctionWithCache() throws Exception {
+    Integer[] key = new Integer[] {1};
+    Map<String, String> properties = Maps.newHashMap();
+    properties.put("lookup-join-cache-size", "3");
+    properties.put("lookup-join-cache-ttl", "10000");
+    initFlinkLookupFunction(properties);
+    List<Row> rows1 = Lists.newArrayList(Row.of(1, "hello"), Row.of(2, "world"), Row.of(3, "foo"));
+
+    List<Row> rows2 =
+        Lists.newArrayList(Row.of(1, "hello2"), Row.of(2, "world2"), Row.of(3, "foo2"));
+
+    RowData expectedRowData = SimpleDataUtil.createRowData(1, "hello");
+    insertTableValues(rows1);
+
+    flinkLookupFunction.open(null);
+    flinkLookupFunction.setCollector(
+        new Collector<RowData>() {
+          @Override
+          public void collect(RowData record) {
+            Assert.assertEquals(
+                "Should join the table data by key correctly. "
+                    + "The joined 'id' value should be '1'",
+                expectedRowData.getInt(0),
+                record.getInt(0));
+            Assert.assertEquals(
+                "Should join the table data by key correctly. "
+                    + "The joined 'data' value should be 'hello'\"",
+                expectedRowData.getString(1),
+                record.getString(1));
+          }
+
+          @Override
+          public void close() {}
+        });
+    flinkLookupFunction.eval(key);
+
+    // Insert into table values "rows2", verify that the function of "cache" is in effect
+    insertTableValues(rows2);
+    flinkLookupFunction.eval(key);
+
+    Thread.sleep(12000);
+    List<RowData> arrayList = Lists.newArrayList();
+    flinkLookupFunction.setCollector(
+        new Collector<RowData>() {
+          @Override
+          public void collect(RowData record) {
+            arrayList.add(record);
+          }
+
+          @Override
+          public void close() {}
+        });
+
+    flinkLookupFunction.eval(key);
+    Assert.assertEquals("Collect data size should be 2", 2, arrayList.size());
+  }
+
+  private void insertTableValues(List<Row> rows) throws Exception {
```

Comment:
if we just want to insert records into Iceberg table, this is probably much slower. we can use `GenericAppenderHelper`

```
    GenericAppenderHelper dataAppender =
        new GenericAppenderHelper(tableResource.table(), FileFormat.PARQUET, TEMPORARY_FOLDER);

    // snapshot0
    List<Record> batch0 =
        RandomGenericData.generate(tableResource.table().schema(), 2, randomSeed.incrementAndGet());
    dataAppender.appendToTable(batch0);
```

Reply:
I have replaced the test case by executing sql